### PR TITLE
Multi pages

### PR DESCRIPTION
**Pull request by marcingrzejszczak**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1013,35 +1061,84 @@
...
+										</sourceDocumentExtensions>
+										<outputDirectory>
+											${project.build.directory}/generated-docs/reference/htmlsingle
+										</outputDirectory>
+										<sourceHighlighter>highlight.js</sourceHighlighter>
+										<doctype>book</doctype>
+										<attributes>
 											<highlightjsdir>js/highlight</highlightjsdir>
 											<highlightjs-theme>atom-one-dark-reasonable</highlightjs-theme>
-											<allow-uri-read>true</allow-uri-read>
```
Comment (Buzzardo) [inline pom.xml]: We need this one for some of the includes (usually code from outside the Asciidoc directory) to be included.

File: pom.xml [hunk 2]
```diff
@@ -1013,35 +1061,84 @@
...
-											<nofooter/>
```
Comment (Buzzardo) [inline pom.xml]: We don't _need_ this one, but we don't need a footer, so please include the nofooter attribute.

File: pom.xml [hunk 3]
```diff
@@ -1013,35 +1061,84 @@
...
-											<toc>left</toc>
```
Comment (Buzzardo) [inline pom.xml]: We need toc: left and toc-levels: 4. They are part of how the dynamic ToC is configured. (4 may be the default for toc-levels, though.)

File: pom.xml [hunk 4]
```diff
@@ -1013,35 +1061,84 @@
...
-											<toc-levels>4</toc-levels>
-											<spring-cloud-version>${project.version}</spring-cloud-version>
-											<sectlinks>true</sectlinks>
```
Comment (Buzzardo) [inline pom.xml]: sectlinks are a nice convenience for people. They enable copying and pasting a link, to send a section to a friend and so on. Please keep them.

File: docs/src/test/bats/ghpages.bats [hunk 5]
```diff
@@ -340,10 +336,9 @@ export -f stubbed_git
 	export GIT_BIN="printing_git"
 	export CURRENT_BRANCH="present"
 	export WHITELISTED_BRANCHES_VALUE="present"
-	export MAIN_ADOC_VALUE="my_doc"
 	cd "${TEMP_DIR}/spring-cloud-stream/"
 	mkdir -p docs/target/generated-docs/
-	touch docs/target/generated-docs/${MAIN_ADOC_VALUE}.html
+	touch docs/target/generated-docs/my_doc.html
```
[the remark this answers is not in the document]
Comment (marcingrzejszczak) [reply docs/src/test/bats/ghpages.bats]: it's just for tests. It no longer gets renamed actually so I've hardcoded the value ;)

File: pom.xml [hunk 6]
```diff
@@ -999,10 +1007,50 @@
 									<artifactId>asciidoctorj-pdf</artifactId>
 									<version>${asciidoctorj-pdf.version}</version>
 								</dependency>
+								<dependency>
```
[the remark this answers is not in the document]
Comment (marcingrzejszczak) [reply pom.xml]: But if it's added then no harm is done, and people can add the diagrams without adding any dependencies. WDYT?

File: pom.xml [hunk 7]
```diff
@@ -1013,35 +1061,84 @@
...
+											<sourceDocumentExtension>.adoc</sourceDocumentExtension>
+											<sourceDocumentExtension>.htmladoc</sourceDocumentExtension>
+										</sourceDocumentExtensions>
+										<outputDirectory>${project.build.directory}/generated-docs/reference/html
+										</outputDirectory>
 										<sourceHighlighter>highlight.js</sourceHighlighter>
 										<doctype>book</doctype>
 										<attributes>
-											// these attributes are required to use the doc resources
-											<docinfo>shared</docinfo>
```
[the remark this answers is not in the document]
Comment (marcingrzejszczak) [reply pom.xml]: Oh, that's an oversight.

File: pom.xml (hunk 2 above)
Comment (marcingrzejszczak) [reply pom.xml]:
```
	<allow-uri-read>true</allow-uri-read>
									<nofooter/>
```

those are just moved to the bottom

**Comment by marcingrzejszczak** [comment]
Added the attributes

```
							<docinfo>shared</docinfo>
									<allow-uri-read>true</allow-uri-read>
									<nofooter/>
									<toc>left</toc>
									<toc-levels>4</toc-levels>
									<sectlinks>true</sectlinks>
```

File: pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Buzzardo) [reply pom.xml]: I think the teams who don't use it won't want the clutter. We have a number of folks who always want to minimize and simplify (bias alert: I'm one of them).

[the remark this answers is not in the document]
Comment (spencergibb) [reply pom.xml]: I don't understand the issue here. It doesn't come in as a project dependency, just a plugin. Spring Cloud core has multiple projects that use it. Those that don't won't notice anything. How will it clutter other projects?

[the remark this answers is not in the document]
Comment (Buzzardo) [reply pom.xml]: I'm not thinking of Spring Cloud. I'm thinking of Boot and Framework and Batch. Marcin has expressed a desire to get these changes into spring-doc-resources, which underpins all those projects. For Spring Cloud, it's up to you folks (and fine by me). I should have been more clear. I was really thinking in terms of the whole collection of Spring docs (as I often do).

[the remark this answers is not in the document]
Comment (spencergibb) [reply pom.xml]: Ah, no worries then.

**Pull request merged**